include/swift/AST/StorageImpl.h

```diff
@@ -355,6 +355,13 @@ class StorageImplInfo {
             getWriteImpl() == WriteImplKind::Immutable);
   }
 
+  bool isSimplyStoredLazy() const {
```

Comment:
This is really checking "is this stored or observed". It doesn't have much to do with being lazy. Maybe add a separete predicate to check for the write impl being 'StoredWithObservers'?